resources/file.go

```diff
@@ -614,6 +660,81 @@ func (obj *FileRes) contentCheckApply(apply bool) (checkOK bool, _ error) {
 	return checkOK, nil
 }
 
+// chmodCheckApply performs a CheckApply for the file permissions.
+func (obj *FileRes) chmodCheckApply(apply bool) (checkOK bool, _ error) {
+	log.Printf("%s[%s]: chmodCheckApply(%t)", obj.Kind(), obj.GetName(), apply)
+
+	if obj.State == "absent" {
+		// File is absent
+		return true, nil
+	}
+
+	if obj.Mode == "" {
+		// No mode specified, everything is ok
+		return true, nil
+	}
+
+	mode, err := obj.mode()
+	if err != nil {
+		return false, err
+	}
+
+	st, err := os.Stat(obj.Path)
+	if err != nil {
+		return false, err
+	}
+
+	if st.Mode() == mode {
+		return true, nil
+	}
+
+	if apply {
+		err := os.Chmod(obj.Path, mode)
+		return false, err
+	}
+
+	return false, nil
+}
+
+// chownCheckApply performs a CheckApply for the file ownership.
+func (obj *FileRes) chownCheckApply(apply bool) (checkOK bool, _ error) {
+	log.Printf("%s[%s]: chownCheckApply(%t)", obj.Kind(), obj.GetName(), apply)
+
+	if obj.State == "absent" {
+		// File is absent or no owner specified
+		return true, nil
+	}
+
+	st, err := os.Stat(obj.Path)
+	if err != nil {
+		return false, err
+	}
+
+	if stUnix, ok := st.Sys().(*syscall.Stat_t); ok {
```

Comment:
`!ok` means mgmt is not running on an unix system. We should probably panic there until we support other operating systems. panic nd not error because it should never happen. This is not a normal error that can occur, but a truly exceptional condition that represents a bug (lack of support for the current platform)